csp/mod.ts

```diff
@@ -0,0 +1,288 @@
+  | "script-src-elem"
+  | "script-src-attr"
+  | "style-src"
+  | "style-src-elem"
+  | "style-src-attr"
+  | "worker-src"
+  | "base-uri"
+  | "plugin-types"
+  | "sandbox"
+  | "form-action"
+  | "frame-ancestors"
+  | "navigate-to"
+  | "report-uri"
+  | "report-to"
+  | "block-all-mixed-content"
+  | "referrer"
+  | "require-sri-for"
+  | "require-trusted-types-for"
+  | "trusted-types"
+  | "upgrade-insecure-requests";
+
+type ReferrerPolicyOptions =
+  | ""
+  | "no-referrer"
+  | "no-referrer-when-downgrade"
+  | "unsafe-url"
+  | "same-origin"
+  | "strict-origin"
+  | "strict-origin-when-cross-origin"
+  | "origin"
+  | "origin-when-cross-origin";
+
+type ExpectCtOptions = {
+  "max-age": number;
+  "enforce"?: boolean;
+  "report-uri"?: string;
+};
+
+type XFrameOptions = "DENY" | "SAMEORIGIN" | boolean | string;
+
+type XPermittedCrossDomainPoliciesOptions =
+  | "none"
+  | "master-only"
+  | "all"
+  | boolean
+  | string;
+
+export interface Options {
+  /** Enforces SSL connections */
+  "Strict-Transport-Security"?: StrictTransportSecurityOptions;
+
+  /** Allows to restrict resources that the web browser loads */
+  "Content-Security-Policy"?: Partial<
+    Record<ContentSecurityPolicyDirectives, string[]>
+  >;
+
+  /** Controls how much referrer information should be included with requests */
+  "Referrer-Policy"?: ReferrerPolicyOptions | ReferrerPolicyOptions[];
+
+  /** Specifies which features of the web browser should be permitted to function */
+  "Permissions-Policy"?: string;
+
+  /** Prevents the use of misissued certificates */
+  "Expect-CT"?: ExpectCtOptions;
+
+  /** Clickjacking protection */
+  "X-Frame-Options"?: XFrameOptions;
+
+  /** MIME sniffing vulnerabilities protection */
+  "X-Content-Type-Options"?: boolean;
+
+  /**  Cross-site scripting (XSS) filter */
+  "X-XSS-Protection"?: boolean;
+
+  /** Restricts loading of Adobe Flash or PDF documents from other domains */
+  "X-Permitted-Cross-Domain-Policies"?: XPermittedCrossDomainPoliciesOptions;
+
+  /** Leaks or fakes information about the server side technology */
+  "X-Powered-By"?: boolean | string;
+}
+
+export const defaults: Options = {
+  "Strict-Transport-Security": {
+    "max-age": DEFAULT_MAX_AGE,
+    "includeSubDomains": true,
+    "preload": true,
+  },
+  "Content-Security-Policy": {
+    "default-src": ["'none'"],
+    "script-src": ["cdn.jsdelivr.net", "'self'"],
+    "style-src": ["cdn.jsdelivr.net", "'self'"],
+    "font-src": ["'self'"],
+    "img-src": ["shield.deno.dev", "'self'"],
+  },
+  "Referrer-Policy": ["no-referrer", "strict-origin-when-cross-origin"],
+  "Permissions-Policy": "camera=(), microphone=(), geolocation=()",
+  "Expect-CT": {
+    "max-age": DEFAULT_MAX_AGE,
+    "enforce": true,
+    "report-uri": "https://localhost:8000/report/",
+  },
+  "X-Frame-Options": true,
+  "X-Content-Type-Options": true,
+  "X-XSS-Protection": true,
+  "X-Permitted-Cross-Domain-Policies": true,
+  "X-Powered-By": "Fake Server",
```

Comment:
The `X-Powered-By` should be disabled by default.

Reply:
Yes, I just added a test value, for testing purpose, to see if it works if a string is set.